### PR TITLE
fix(livereload): support HTTP/2 --h2 option

### PR DESCRIPTION
**Pull request by attila-boros**
Fix for #510

### CODE REVIEW

**Comment by attila-boros** [comment]
The CI build is failing with _ERROR: "test:wdi5" exited with 1_. @petermuessig could you please take a look at this? It seems unrelated to my PR.

**Comment by vobu** [comment]
> The CI build is failing with _ERROR: "test:wdi5" exited with 1_. @petermuessig could you please take a look at this? It seems unrelated to my PR.

yep, can confirm, this is unrelated to you PR. will take a closer look once i have time (haha).

File: packages/ui5-middleware-livereload/lib/livereload.js [hunk 1]
```diff
@@ -67,13 +67,31 @@ module.exports = async ({ resources, options }) => {
...
+    let serverOptions = {
+        debug: debug,
+        extraExts: extraExts ? extraExts.split(",") : undefined,
+        port: port,
+        exclusions: exclusions
+    };
+    
+    const cli = require("yargs");
+    if (cli.argv.h2) {
+        const path = require("path");
```
Comment (vobu) [inline packages/ui5-middleware-livereload/lib/livereload.js:80]: `path` is already required https://github.com/ui5-community/ui5-ecosystem-showcase/blob/8f1ff9d76a5f8ef85db4af62748cc7f5ec456011/packages/ui5-middleware-livereload/lib/livereload.js#L3, so no need to do it again

File: packages/ui5-middleware-livereload/lib/livereload.js [hunk 2]
```diff
@@ -67,13 +67,31 @@ module.exports = async ({ resources, options }) => {
...
+        const os = require("os");
+        const fs = require("fs");
+        
+        sslKeyPath = cli.argv.key ? cli.argv.key : path.join(os.homedir(), ".ui5", "server", "server.key");
+        sslCertPath = cli.argv.cert ? cli.argv.cert : path.join(os.homedir(), ".ui5", "server", "server.crt");
+
```
Comment (vobu) [inline packages/ui5-middleware-livereload/lib/livereload.js:86]: please provide a `log.debug(...)` on where the runtime retrieves the certs from - convenient for the user when things break

**Review by vobu — CHANGES_REQUESTED** [review]
thanks for the PR 🙌 

couple of small changes please.
Plus: could you update the documentation (`README.md`), briefly explaining how to use the middleware w/ http2? Most relevant might be how to pass the ssl cert file location to the runtime. Thanks!

File: packages/ui5-middleware-livereload/lib/livereload.js (hunk 1 above)
Comment (attila-boros) [reply packages/ui5-middleware-livereload/lib/livereload.js]: Good catch, I didn't observe it.

File: packages/ui5-middleware-livereload/lib/livereload.js (hunk 2 above)
Comment (attila-boros) [reply packages/ui5-middleware-livereload/lib/livereload.js]: Debug info added.

**Comment by attila-boros** [comment]
I updated README.md, the idea is to use the same key and certificate as the UI5 server: https://sap.github.io/ui5-tooling/pages/CLI/#ui5-serve

If the --h2 option is set, the middleware checks if --key and --cert are set, and takes the key and certificate from those options, or it falls back to the default key and cert in ~/.ui5/server.

Currently I'm using the default key and certificate generated automatically by the server.

**Review by vobu — APPROVED** [review]
cool shakes!

**Comment by petermuessig** [comment]
```
Successfully published:
 - ui5-middleware-livereload@0.5.3
```

**Pull request merged**